app/src/main/java/com/spoony/spoony/core/designsystem/theme/Theme.kt

```diff
@@ -1,57 +1,317 @@
+    White: Color = white,
+    Black: Color = black,
+    Gray0: Color = gray0,
+    Gray100: Color = gray100,
+    Gray200: Color = gray200,
+    Gray300: Color = gray300,
+    Gray400: Color = gray400,
+    Gray500: Color = gray500,
+    Gray600: Color = gray600,
+    Gray700: Color = gray700,
+    Gray800: Color = gray800,
+    Gray900: Color = gray900,
+) = SpoonyColors(
+    Main0,
+    Main100,
+    Main200,
+    Main300,
+    Main400,
+    Main500,
+    Main600,
+    Main700,
+    Main800,
+    Main900,
+    Orange400,
+    Pink400,
+    Green400,
+    Blue400,
+    Purple400,
+    Orange100,
+    Pink100,
+    Green100,
+    Blue100,
+    Purple100,
+    Error400,
+    White,
+    Black,
+    Gray0,
+    Gray100,
+    Gray200,
+    Gray300,
+    Gray400,
+    Gray500,
+    Gray600,
+    Gray700,
+    Gray800,
+    Gray900,
+    isLight = true
 )
 
+private val LocalSpoonyColors = staticCompositionLocalOf<SpoonyColors> {
+    error("No SpoonyColors provided")
+}
+
+private val LocalSpoonyTypography = staticCompositionLocalOf<SpoonyTypography> {
+    error("No SpoonyTypography provided")
+}
+
+/*
+* SpoonyTheme
+*
+* Color에 접근하고 싶을때 SpoonyTheme.colors.primary 이런식으로 접근하면 됩니다.
+* Typo를 변경하고 싶다면 SpoonyTheme.typography.heading48B 이런식으로 접근하면 됩니다.
+* */
+object SpoonyAndroidTheme {
+    val colors: SpoonyColors
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalSpoonyColors.current
+
+    val typography: SpoonyTypography
+        @Composable
+        @ReadOnlyComposable
+        get() = LocalSpoonyTypography.current
+
+}
+
 @Composable
-fun SpoonyAndroidTheme(
-    darkTheme: Boolean = isSystemInDarkTheme(),
-    // Dynamic color is available on Android 12+
-    dynamicColor: Boolean = true,
+fun ProvideSpoonyColorsAndTypography(
+    colors: SpoonyColors,
+    typography: SpoonyTypography,
     content: @Composable () -> Unit
 ) {
-    val colorScheme = when {
-        dynamicColor && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
-            val context = LocalContext.current
-            if (darkTheme) dynamicDarkColorScheme(context) else dynamicLightColorScheme(context)
-        }
-
-        darkTheme -> DarkColorScheme
-        else -> LightColorScheme
-    }
-
-    MaterialTheme(
-        colorScheme = colorScheme,
-        typography = Typography,
+    val provideColors = remember { colors.copy() }
+    provideColors.update(colors)
+    val provideTypography = remember { typography.copy() }.apply { update(typography) }
+    CompositionLocalProvider(
+        LocalSpoonyColors provides provideColors,
+        LocalSpoonyTypography provides provideTypography,
         content = content
     )
 }
+
+@Composable
+fun SpoonyAndroidTheme(darkTheme: Boolean = false, content: @Composable () -> Unit) {
+    val colors = SpoonyLightColors()
+    val typography = SpoonyTypography()
+    ProvideSpoonyColorsAndTypography(colors, typography) {
+        MaterialTheme(content = content)
+    }
+}
```

Comment:
P2: Provider 함수가 Colors와 Typography를 함께 처리하여 유연성이 부족한 것 같아요. 이 부분도 아티클 참고하시면 될 것 같습니다.
예시는 아래같이 작성하면 좋을 것 같아요.
```kotlin
@Composable
fun SpoonyTheme(
    darkTheme: Boolean = false,
    colors: SpoonyColors = defaultSpoonyColors,
    typography: SpoonyTypography = defaultSpoonyTypography,
    content: @Composable () -> Unit
) {
    ProvideSpoonyTheme(
        colors = colors,
        typography = typography
    ) {
        MaterialTheme(content = content)
    }
}
```

P5: 'Android'라는 이름을 붙이는게 불필요한 플랫폼 지정자라고 생각해요 그래서 제거하는게 좋은것 같아요. 다른 분들 생각은 어떠실지..!

Reply:
이것도 현재 상황에서는 맞는 이야기에요! 하지만 추후 다크모드를 가정하게 된다면 지금과 같은 방식을 사용하는 것이 좋아보입니다.

다크모드를 적용하게 된다면 코드가 아래처럼 바뀌게 될거에요
```suggestion

@Composable
fun SpoonyAndroidTheme(
    darkTheme: Boolean = isSystemInDarkTheme(), 
    content: @Composable () -> Unit
) {
    val colors = if (darkTheme) spoonyDarkColors() else spoonyLightColors()
    val typography = SpoonyTypography()
    ProvideSpoonyColorsAndTypography(colors, typography) {
        MaterialTheme(content = content)
    }
}
```

typography의 경우 변경될 일이 없을 것 같아서 제시해주신 방법대로 하는 것이 더 좋을 것 같네요!!



Android prefix의 경우에도 없는것이 더 좋아보입니다! 꼼꼼한 리뷰 최고~

Reply:
좋은 리뷰 감사합니다..!